client/edge.go

```diff
@@ -53,12 +60,39 @@ type EdgeClient interface {
 
 // NewEdgeConnector create EdgeConnector with the given BrokerConfig for the given EdgeClient
 func NewEdgeConnector(cfg *BrokerConfig, ecl EdgeClient) (*EdgeConnector, error) {
+	var certificates []tls.Certificate
+	var caCertPool *x509.CertPool
+	if len(cfg.DeviceCert) > 0 {
+		keyPair, err := tls.LoadX509KeyPair(cfg.DeviceCert, cfg.DeviceKey)
+		if err != nil {
+			logger.Errorf("error reading client certificate files - \"%s, %s\"", cfg.DeviceCert, cfg.DeviceKey)
+			return nil, err
```

Comment:
Error log and return is very bad practice because the caller can duplicate the log.